roles/nginxplus/files/conf/http/cdh_prod_geniza.conf

```
@@ -60,8 +60,11 @@ server {
         proxy_set_header X-Real-IP $remote_addr;
         proxy_cache geniza_prodcache;
 	# upgrade connections
-	proxy_set_header Upgrade $http_upgrade;
-	proxy_set_header Connection $connection_upgrade;
+	      proxy_set_header Upgrade $http_upgrade;
+	      proxy_set_header Connection $connection_upgrade;
+        proxy_connect_timeout      2h;
```

Just in case the indentation here was unintentional and you want to clean it up:

```suggestion
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
        proxy_connect_timeout      2h;
```